src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1435,6 +1437,15 @@ public Image getResolutionVariant(double width, double height) {
                     }
                 }
             }
+            if (retVal.getWidth(null) != w) {
```

Comment:
It isn't in this webrev but i jyst noticed
            // We only care about width since we don't support non-rectangular icons
I think that meant non-square ???

Reply:
> It isn't in this webrev but i jyst noticed
> // We only care about width since we don't support non-rectangular icons
> I think that meant non-square ???

Yes. Now it seems to be apparent that English is not my native language...